app/lib/submission_builder/ty2022/states/authentication_header.rb

```diff
@@ -63,19 +62,17 @@ def xml_type_for_state_id(state_id)
         def state_id_to_xml(state_id, xml_builder)
           xml_type = xml_type_for_state_id(state_id)
           if xml_type
-            xml_builder.send("#{xml_type}Num", state_id.id_number)
-            xml_builder.send("#{xml_type}StCd", state_id.state)
+            xml_builder.send("#{xml_type}Num", state_id.id_number) if state_id.id_number.present?
+            xml_builder.send("#{xml_type}StCd", state_id.state) if state_id.state.present?
             xml_builder.send("#{xml_type}ExprDt") do
               if state_id.non_expiring?
-                xml_builder.NonExpr
+                xml_builder.NonExpr "X"
```

Comment:
I looked at the XSD for this:
```<xsd:element name="NonExpr" fixed="Non-Expiring"/>```
It looks to me that this should have the value "Non-Expiring" rather than "X"?